html/modules/custom/reliefweb_rivers/src/Services/TopicRiver.php

```diff
@@ -0,0 +1,210 @@
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $bundle = 'topic';
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPageTitle() {
+    return $this->t('Topics');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getViews() {
+    return [
+      'all' => $this->t('All Topics'),
+    ];
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Pager\PagerManagerInterface $pager_manager
+   *   The pager manager service.
+   * @param \Drupal\Core\Pager\PagerParametersInterface $pager_parameters
+   *   The pager parameter service.
+   * @param \Drupal\reliefweb_api\Services\ReliefWebApiClient $api_client
+   *   The ReliefWeb API Client service.
+   * @param \Drupal\Core\StringTranslation\TranslationInterface $string_translation
+   *   The translation manager service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   */
+  public function __construct(PagerManagerInterface $pager_manager, PagerParametersInterface $pager_parameters, ReliefWebApiClient $api_client, TranslationInterface $string_translation, EntityTypeManagerInterface $entity_type_manager) {
+    $this->pagerManager = $pager_manager;
+    $this->pagerParameters = $pager_parameters;
+    $this->apiClient = $api_client;
+    $this->stringTranslation = $string_translation;
+    $this->url = static::getRiverUrl($this->bundle);
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRiverContent() {
+    $page = $this->pagerParameters->findPage();
+    $offset = $page * $this->limit;
+    $totalCount = 0;
+
+    $query = $this->entityTypeManager->getStorage('node')->getQuery()
+      ->condition('type', 'topic')
+      ->condition('status', 1)
+      ->sort('created', 'DESC');
+
+    $group = $query->orConditionGroup()
+      ->notExists('field_bury')
+      ->condition('field_bury', 0, '<>');
+    $query->condition($group);
+    $totalCount = $query->count()->execute();
+
+    $query = $this->entityTypeManager->getStorage('node')->getQuery()
+      ->condition('type', 'topic')
+      ->condition('status', 1)
+      ->sort('created', 'DESC');
+
+    $group = $query->orConditionGroup()
+      ->notExists('field_bury')
+      ->condition('field_bury', 0, '<>');
+    $query->condition($group);
+    $nids = $query->range($offset, $this->limit)
+      ->execute();
+    $topics = $this->entityTypeManager->getStorage('node')->loadMultiple($nids);
+
+    $reliefweb_topics = [];
+    foreach ($topics as $nid => $topic) {
+      $key = $topic->field_featured->value . '_' . $nid;
+      $icon = [];
+
+      if (!$topic->field_icon->isEmpty() && $topic->field_icon->entity) {
+        $icon = [
+          'uri' => file_create_url($topic->field_icon->entity->getFileUri()),
```

Comment:
```suggestion
          'uri' => $topic->field_icon->entity->getFileUri(),
```
The code for the image style in the topic river template expects a URI.